### ISSUE TITLE
Question about inferring

### ISSUE THREAD

**Issue by ten10second** [TiagoCortinhal/TITAN-Next#4]
Thank you for your reply. I've also noticed this inference process. Based on this part of your code, I've written a user.py file to reproduce the results using the pre - trained model and save the segmentation results of visible light images generated by the generator.

`  def infer_subset(self, loader):

    # switch to evaluate mode
    self.model.eval()
    # empty the cache to infer in high res
    if self.gpu:
      torch.cuda.empty_cache()

    with torch.no_grad():
      for i, (_, _, _, _, proj, _, proj_labels_one_hot, _,path_seq_1,path_name_1) in enumerate(loader):

        if self.gpu:
          proj = proj.cuda()
          # proj_labels_one_hot =  proj_labels_one_hot.cuda()
          print("=======")
          print(proj.size(), proj_labels_one_hot.size())
          
          generated_segimg, fakedepth, bn= self.model(proj,  proj_labels_one_hot)
)
          generated_segimg = nn.Softmax(dim=1)(generated_segimg)
          generated_argmax = generated_segimg.argmax(dim=1, keepdim=True).cpu().numpy()

          #self.evaluator.addBatch(generated_argmax,rgb_labels)
          generated_argmax = colorize_mask(generated_argmax[0][0]).convert(mode='RGB')
          convertion_dict = {0: 0,
                              1: 26,
                              2: 33,
                              3: 32,
                              4: 27,
                              5: 29,
                              6: 24,
                              7: 7,
                              8: 8,
                              9: 11,
                              10: 13,
                              11: 21,
                              12: 22,
                              13: 17,
                              14: 20}
          #fakedepth = nn.Upsample(size=(376, 1241), mode='bilinear', align_corners=True)(fakedepth)
          print(generated_argmax[0][0].shape)
          # newArray2 = np.copy(generated_argmax)  # .astype('uint8')
          # for k, v in convertion_dict.items():
          #     newArray2[generated_argmax[0][0] == k] = v

          save_path = os.path.join(self.logdir, "sequences",
                        path_seq_1, "predictions", "{:06}.png".format(path_name_1[0].split(".")[0]))

          generated_argmax.save(save_path)
`
I have the following questions:
1. As can be seen from the code of the following generator network model, the dimension of `proj_labels_one_hot` is increased through the convolution operation. According to the data shapes printed in the `user.py` script, the data dimensions of `proj` and `proj_labels_one_hot` are `torch.Size([1, 5, 64, 512])` and `torch.Size([1, 20, 64, 512])` respectively. After stacking these two data along the first dimension, the number of channels does not equal 5.
2. In the `--split test` mode, `self.gt = False`. At this time, `proj_labels_one_hot = []`. My confusion is whether the input of the generator model does not include `proj_labels_one_hot` in the test mode. 

I'm very interested in your work. I've been reading and trying to successfully run your code recently. Thank you again for your reply and your excellent work.

**Comment by alikesierzhao0411** [comment]
> Thank you for your reply. I've also noticed this inference process. Based on this part of your code, I've written a user.py file to reproduce the results using the pre - trained model and save the segmentation results of visible light images generated by the generator.
> 
> ` def infer_subset(self, loader):
> 
> ```
> # switch to evaluate mode
> self.model.eval()
> # empty the cache to infer in high res
> if self.gpu:
>   torch.cuda.empty_cache()
> 
> with torch.no_grad():
>   for i, (_, _, _, _, proj, _, proj_labels_one_hot, _,path_seq_1,path_name_1) in enumerate(loader):
> 
> 
>     if self.gpu:
>       proj = proj.cuda()
>       # proj_labels_one_hot =  proj_labels_one_hot.cuda()
>       print("=======")
>       print(proj.size(), proj_labels_one_hot.size())
>       
>       generated_segimg, fakedepth, bn= self.model(proj,  proj_labels_one_hot)
> ```
> 
> ) generated_segimg = nn.Softmax(dim=1)(generated_segimg) generated_argmax = generated_segimg.argmax(dim=1, keepdim=True).cpu().numpy()
> 
> ```
>       #self.evaluator.addBatch(generated_argmax,rgb_labels)
>       generated_argmax = colorize_mask(generated_argmax[0][0]).convert(mode='RGB')
>       convertion_dict = {0: 0,
>                           1: 26,
>                           2: 33,
>                           3: 32,
>                           4: 27,
>                           5: 29,
>                           6: 24,
>                           7: 7,
>                           8: 8,
>                           9: 11,
>                           10: 13,
>                           11: 21,
>                           12: 22,
>                           13: 17,
>                           14: 20}
>       #fakedepth = nn.Upsample(size=(376, 1241), mode='bilinear', align_corners=True)(fakedepth)
>       print(generated_argmax[0][0].shape)
>       # newArray2 = np.copy(generated_argmax)  # .astype('uint8')
>       # for k, v in convertion_dict.items():
>       #     newArray2[generated_argmax[0][0] == k] = v
> 
>       save_path = os.path.join(self.logdir, "sequences",
>                     path_seq_1, "predictions", "{:06}.png".format(path_name_1[0].split(".")[0]))
> 
> 
>       generated_argmax.save(save_path)
> ```
> 
> ` I have the following questions:
> 
> 1. As can be seen from the code of the following generator network model, the dimension of `proj_labels_one_hot` is increased through the convolution operation. According to the data shapes printed in the `user.py` script, the data dimensions of `proj` and `proj_labels_one_hot` are `torch.Size([1, 5, 64, 512])` and `torch.Size([1, 20, 64, 512])` respectively. After stacking these two data along the first dimension, the number of channels does not equal 5.
> 2. In the `--split test` mode, `self.gt = False`. At this time, `proj_labels_one_hot = []`. My confusion is whether the input of the generator model does not include `proj_labels_one_hot` in the test mode.
> 
> I'm very interested in your work. I've been reading and trying to successfully run your code recently. Thank you again for your reply and your excellent work.

Hi, I am also troubled by the same problem when running this work. Do you have any progress to share? I also hope we can discuss this work.
